src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -210,15 +210,19 @@ EmberAfStatus OnOffServer::setOnOffValue(chip::EndpointId endpoint, uint8_t comm
         {
             emberAfOnOffClusterLevelControlEffectCallback(endpoint, newValue);
         }
-#endif
-
-        // write the new on/off value
-        status = Attributes::OnOff::Set(endpoint, newValue);
-        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        else
         {
-            emberAfOnOffClusterPrintln("ERR: writing on/off %x", status);
-            return status;
+#endif
+            // write the new on/off value
+            status = Attributes::OnOff::Set(endpoint, newValue);
+            if (status != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                emberAfOnOffClusterPrintln("ERR: writing on/off %x", status);
+                return status;
+            }
+#ifdef EMBER_AF_PLUGIN_LEVEL_CONTROL
```

Comment:
I am also not a fan of the use of those ember_af_plugin defines. I know there is still multiple of those throughout the code but they come from zigbee

We now have means check the metadat of the cluster and use the featuremap to determine if a feature should be used or not.

@bzbarsky-apple what do you think? I'll differ to your call on this

Reply:
I think the issue is that this code is calling functions that may not even exist if the level control thing is not compiled in....

But past that, doing feature checks seems fine if there is a feature bit for this.